homeassistant/components/azure_event_hub/__init__.py

```diff
@@ -1,89 +1,240 @@
 """Support for Azure Event Hubs."""
+import asyncio
 import json
 import logging
+import queue
+import threading
+import time
 from typing import Any, Dict
 
-from azure.eventhub import EventData, EventHubClientAsync
+from azure.eventhub import (
+    EventData,
+    EventDataBatch,
+    EventHubProducerClient,
+    EventHubSharedKeyCredential,
+)
+from azure.eventhub.exceptions import EventHubError
 import voluptuous as vol
 
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import (
     EVENT_HOMEASSISTANT_STOP,
-    EVENT_STATE_CHANGED,
+    MATCH_ALL,
     STATE_UNAVAILABLE,
     STATE_UNKNOWN,
 )
-from homeassistant.core import Event, HomeAssistant
+from homeassistant.core import Event, HomeAssistant, callback
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entityfilter import FILTER_SCHEMA
 from homeassistant.helpers.json import JSONEncoder
 
-_LOGGER = logging.getLogger(__name__)
+from .const import (
+    CONF_EVENT_HUB_CON_STRING,
+    CONF_EVENT_HUB_INSTANCE_NAME,
+    CONF_EVENT_HUB_NAMESPACE,
+    CONF_EVENT_HUB_SAS_KEY,
+    CONF_EVENT_HUB_SAS_POLICY,
+    CONF_FILTER,
+    DOMAIN,
+)
 
-DOMAIN = "azure_event_hub"
+_LOGGER = logging.getLogger(__name__)
 
-CONF_EVENT_HUB_NAMESPACE = "event_hub_namespace"
-CONF_EVENT_HUB_INSTANCE_NAME = "event_hub_instance_name"
-CONF_EVENT_HUB_SAS_POLICY = "event_hub_sas_policy"
-CONF_EVENT_HUB_SAS_KEY = "event_hub_sas_key"
-CONF_FILTER = "filter"
+QUEUE_BACKLOG_SECONDS = 30
+PAUSE_WHEN_EMPTY = 5
+BATCH_TIMEOUT = 1
 
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.Schema(
             {
-                vol.Required(CONF_EVENT_HUB_NAMESPACE): cv.string,
-                vol.Required(CONF_EVENT_HUB_INSTANCE_NAME): cv.string,
-                vol.Required(CONF_EVENT_HUB_SAS_POLICY): cv.string,
-                vol.Required(CONF_EVENT_HUB_SAS_KEY): cv.string,
-                vol.Required(CONF_FILTER): FILTER_SCHEMA,
-            }
+                vol.Optional(CONF_EVENT_HUB_CON_STRING, ""): cv.string,
+                vol.Optional(CONF_EVENT_HUB_NAMESPACE, ""): cv.string,
+                vol.Optional(CONF_EVENT_HUB_INSTANCE_NAME, ""): cv.string,
+                vol.Optional(CONF_EVENT_HUB_SAS_POLICY, ""): cv.string,
+                vol.Optional(CONF_EVENT_HUB_SAS_KEY, ""): cv.string,
```

Comment:
remove the default of `""` or set it with DEFAULT_ constants if you realy need this.